conans/client/runner.py

```diff
@@ -100,3 +102,27 @@ def _simple_os_call(self, command, cwd):
             finally:
                 os.chdir(old_dir)
             return result
+
+
+if getattr(sys, 'frozen', False) and 'LD_LIBRARY_PATH' in os.environ:
+
+    # http://pyinstaller.readthedocs.io/en/stable/runtime-information.html#ld-library-path-libpath-considerations
+    pyinstaller_bundle_dir = os.environ['LD_LIBRARY_PATH'].replace(
+        os.environ.get('LD_LIBRARY_PATH_ORIG', ''), ''
+    ).strip(';:')
+
+    @contextmanager
+    def pyinstaller_bundle_env_cleaned():
+        """Removes the pyinstaller bundle directory from LD_LIBRARY_PATH
+
+        :return: None
+        """
+        ld_library_path = os.environ['LD_LIBRARY_PATH']
+        os.environ['LD_LIBRARY_PATH'] = ld_library_path.replace(pyinstaller_bundle_dir, '')
```

Comment:
Could this leave trailing ;: too?

Reply:
yes, this could leave a double or trailing pathsep. However, I have not run into problems with those.
To be safe we could do a full `os.pathsep.join(p for p in ld_library_path.split(os.pathsep) if p != pyinstaller_bundle_dir)` instead.